src/main/java/com/wanderersoftherift/wotr/datagen/ModAbilityTagsProvider.java

```diff
@@ -0,0 +1,28 @@
+package com.wanderersoftherift.wotr.datagen;
+
+import com.wanderersoftherift.wotr.WanderersOfTheRift;
+import com.wanderersoftherift.wotr.abilities.AbstractAbility;
+import com.wanderersoftherift.wotr.init.ModTags;
+import com.wanderersoftherift.wotr.init.RegistryEvents;
+import net.minecraft.core.HolderLookup;
+import net.minecraft.data.PackOutput;
+import net.minecraft.data.tags.TagsProvider;
+import net.minecraft.resources.ResourceKey;
+
+import java.util.concurrent.CompletableFuture;
+
+public class ModAbilityTagsProvider extends TagsProvider<AbstractAbility> {
+    // Get parameters from the `GatherDataEvent`s.
+    public ModAbilityTagsProvider(PackOutput output, CompletableFuture<HolderLookup.Provider> lookupProvider) {
+        // Second parameter is the registry key we are generating the tags for.
+        super(output, RegistryEvents.ABILITY_REGISTRY, lookupProvider, WanderersOfTheRift.MODID);
+    }
+
+    @Override
+    protected void addTags(HolderLookup.Provider lookupProvider) {
+        tag(ModTags.Abilities.RIFT_DROPS)
```

Comment:
Looks like
```
        tag(ModTags.Abilities.RIFT_DROPS)
                .add(TagEntry.optionalElement(WanderersOfTheRift.id("dash")))
                .add(TagEntry.optionalElement(WanderersOfTheRift.id("fireball")))
                .add(TagEntry.optionalElement(WanderersOfTheRift.id("heal")));
```
works without the abilities needing to be data gen (but do need the ability registry bootstrapped).

Reply:
Experimenting locally, by combining this with using the DeferredHolder approach for the recipes and changing the DeferrableRegistryCodec to eliminate existence checks when encoding, it is possible for data gen to reference non-datagenned abilities. It would be nice if we could set the DataComponent to use different codecs when being used in game to being used in data gen though, so we could keep the existence checks when encoding in game.